lib/Transforms/Secretize/WrapGeneric.cpp

```diff
@@ -20,13 +20,43 @@ struct WrapWithGeneric : public OpRewritePattern<func::FuncOp> {
   LogicalResult matchAndRewrite(func::FuncOp op,
                                 PatternRewriter &rewriter) const override {
     bool hasSecrets = false;
+    bool secretHasUse = false;
 
+    // check if any of the arguments are secret
+    // and if any of the secret arguments are used
+    for (unsigned i = 0; i < op.getNumArguments(); i++) {
```

Comment:
If we proceed with this PR, I think this could be made slightly cleaner by keeping the "!hasSecrets" check from the original code, and replacing this loop with

``` 
for (unsigned i = 0; i < op.getNumArguments(); i++) {
  if (op.getArgAttr(i, secret::SecretDialect::kArgSecretAttrName) !=
      nullptr && !op.getArgument(i).getUses().empty()) {
    op.removeArgAttr(i, secret::SecretDialect::kArgSecretAttrName);
  }
}
```

That is, always remove secret attributes for unused values, and then the original check for "no secret values" would work as normal.